### PR TITLE
Improve MAVEN build Performance

### PR DESCRIPTION
**Pull request by hongdongni**
Maven allows you to run the compiler as a separate process by setting `<fork>true</fork>`. This feature can lead to much less garbage collection and make Maven build faster. This project has more than 1000 source files. We can consider enabling this feature.

Maven can recompile only the classes that were affected by a change. This feature is the default. We can activate it by setting `<useIncrementalCompilation>true</useIncrementalCompilation>`.

=====================
If there are any inappropriate modifications in this PR, please give me a reply and I will change them.

### CODE REVIEW

**Comment by dkayiwa** [comment]
Thanks @hongdongni for this contribution!
Did you try compile with these changes and notice a build performance? If yes, from what build time to what? Or what percentage improvement?

File: pom.xml [hunk 1]
```diff
@@ -583,6 +583,8 @@
 						<target>${javaCompilerVersion}</target>
 						<source>${javaCompilerVersion}</source>
 						<encoding>UTF-8</encoding>
+						<fork>true</fork>
+						<useIncrementalCompilation>true</useIncrementalCompilation>
```
Comment (HerbertYiga) [inline pom.xml:587]: @hongdongni  and the second thing  is to confirm whether maven doesnt  set useIncrementalCompilation  to true by default

Comment (hongdongni) [reply pom.xml:587]: I'm going to delete this line

**Comment by hongdongni** [comment]
Hi @dkayiwa !
| fork | true | false(by default)|
| ---|---|---|
|time| [41m 13s](https://github.com/openmrs/openmrs-core/actions/runs/1541683700)|[1h 11m 48s](https://github.com/hongdongni/openmrs-core/actions/runs/1548059316)|

**Pull request closed without merging**